app/models/ingest_measurement.rb

```diff
@@ -11,5 +11,129 @@ class << self
     def data_for(query)
       search(query: query).results.map(&:_source)
     end
+
+    def query_sensor_data(device_urn) # rubocop:disable Metrics/MethodLength
+      search "query":
+                 { "bool": {
+                   "filter": [
+                     {
+                       "range": {
+                         "@timestamp": {
+                           "gte": 'now-8w',
+                           "lte": 'now'
+                         }
+                       }
+                     },
+                     {
+                       "term": {
+                         "device_urn": device_urn
+                       }
+                     }
+                   ]
+                 } },
+             "aggs": {
+               "sensor_data": {
+                 "date_histogram": {
+                   "interval": '12h',
+                   "field": '@timestamp',
+                   "min_doc_count": 0,
+                   "extended_bounds": {
+                     "min": 'now-8w',
+                     "max": 'now'
+                   },
+                   "format": 'epoch_millis'
```

Comment:
If you use this `format` instead,

```suggestion
                   "format": 'yyyy-MM-dd HH'
```

you can use `bucket['key_as_string']` instead on line 97 of `app/helpers/device_stories_helper.rb`.